File: ios-samples/KiipObjCSample/KiipObjCSample/KiipTableViewController.m
//
//  KiipTableViewController.m
//  KiipObjCSample
//
//  Copyright © 2018 Kiip. All rights reserved.
//

#import "KiipTableViewController.h"
#import <KiipSDK/KiipSDK.h>

@interface KiipTableViewController ()

@property (strong, nonatomic) KPTableView *placer;
@property (nonatomic,strong) NSMutableArray *data;

@end

@implementation KiipTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.title = @"TableView";
    CGRect tableRect = CGRectMake(0, 50, self.view.bounds.size.width, self.view.bounds.size.height - 50);
    self.tableview = [[UITableView alloc] initWithFrame:tableRect];
    [self.tableview registerClass:[UITableViewCell class] forCellReuseIdentifier:@"cell"];
    [self.tableview kp_setDelegate:self];
    [self.tableview kp_setDataSource:self];

    self.data = [NSMutableArray arrayWithCapacity:1000];
    for (int i = 0; i < 1000; i++) {
        [self.data addObject:[NSString stringWithFormat:@"%d", i]];
    }
    self.placer = [KPTableView placerWithTableView:self.tableview viewController:self];
    
    [self.view addSubview:self.tableview];
    [self.placer loadAdsForAdUnitID:@"Enter Your Moment ID"];

}

#pragma mark - UITableViewDataSource
- (NSInteger)numberOfSectionsInTableView:(UITableView *)theTableView
{
    return 1;   //have to think about when a user define more then 1 section
}


- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return [self.data count];
}


- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
{
    static NSString *cellIdentifier = @"kiip";
    UITableViewCell *cell = (UITableViewCell *)[tableView dequeueReusableCellWithIdentifier:cellIdentifier];
    
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
    }
    
    cell.textLabel.text =  [self.data objectAtIndex:indexPath.row];
    //    cell.textLabel.text = [NSString stringWithFormat:@"Cell : %ld", (long)indexPath.row];
    return cell;
    
}

- (CGFloat) tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return UITableViewAutomaticDimension;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: ios-samples/KiipSwiftSample/KiipSwiftSample/KiipTableViewController.swift
//
//  KiipTableViewController.swift
//  KiipSwiftSample
//
//  Copyright © 2018 Kiip. All rights reserved.
//

import UIKit
import KiipSDK

class KiipTableViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {
    var tableview: UITableView?
    var placer: KPTableView?
    var data: [String] = []
    
    override func viewDidLoad() {
        super.viewDidLoad()
        title = "TableView"
        let tableRect = CGRect(x: 0, y: 50, width: view.bounds.size.width, height: view.bounds.size.height - 50)
        tableview = UITableView(frame: tableRect)
        tableview?.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
        tableview?.kp_setDelegate(self)
        tableview?.kp_setDataSource(self)
        

        for i in 0..<1000 {
            data.append("\(i)")
        }
        placer = KPTableView.placer(with: tableview, viewController: self)
        
        view.addSubview(tableview!)
        placer?.loadAds(forAdUnitID: "d")

        // Do any additional setup after loading the view.
    }
    
    // MARK: - UITableViewDataSource
    func numberOfSections(in theTableView: UITableView) -> Int {
        return 1 //have to think about when a user define more then 1 section
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return data.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cellIdentifier = "kiip"
        var cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier)
        
        if cell == nil {
            cell = UITableViewCell(style: .default, reuseIdentifier: cellIdentifier)
        }
        cell?.textLabel?.text = self.data[indexPath.row]
        return cell!
        
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return UITableViewAutomaticDimension
    }
        //

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
